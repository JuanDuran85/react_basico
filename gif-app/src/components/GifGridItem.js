import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({id,title,url}) => {
    return (
        <div className="card animate__animated animate__swing">
            <img src={url} title={title} alt={id}/>
            <p>{title}</p>
        </div>
    );
};

GifGridItem.propTypes = {
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};