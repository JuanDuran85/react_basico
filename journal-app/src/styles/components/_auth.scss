.auth {
    &__main {
        align-items: center;
        background-color: $primary;
        display: flex;
        justify-content: center;
        margin: 0;
        height: 100vh;
        width: 100vw;
    }

    &__box-container {
        background-color: $white;
        box-shadow: 0px 3px $dark-gray;
        padding: 20px;
        width: 250px;
    }

    &__title {
        color: darken($color: $primary, $amount: 20);
        margin-bottom: 20px;
    }

    &__input {
        color: $dark-gray;
        border: 0px;
        border-bottom: 1px solid $light-gray;
        font-size: 16px;
        margin-bottom: 10px;
        height: 20px;
        width: 100%;
        transition: border-bottom .3s ease;

        &:focus {
            border-bottom: 1px solid $primary;
            outline: none;
        }
    }

    &__social-network {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        padding-bottom: 20px;
        padding-top: 20px;
        width: 100%;
    }

    &__alert-error {
        background-color: red;
        border-radius: 5px;
        color: $white;
        display: flex;
        justify-content: center;
        margin-bottom: 10px;
        padding: 5px;
    }
}