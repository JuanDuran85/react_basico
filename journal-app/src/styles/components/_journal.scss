.journal {
    &__main-content{
        display: flex;
    }

    &__sidebar {
        background-color: $dark-gray;
        color: $white;
        display: flex;
        flex-direction: column;
        height: 100%;
        padding-left: 10px;
        padding-right: 10px;
        width: 450px;
    }

    &__sidebar-navbar {
        display: flex;
        justify-content: space-between;

        h3 {
            font-weight: lighter;
        }
    }

    &__new-entry {
        align-items: center;
        cursor: pointer;
        display: flex;
        flex-direction: column;
        justify-content: center;
        margin-top: 30px;
        width: 100%;
        transition: color .3s ease;
        &:hover {
            color: darken($color: $white, $amount: 20);
        }
    }

    &__entries {
        flex: 1 1 auto;
        margin-top: 30px;
        overflow: scroll; 
    }

    &__entry {
        background-color: $white;
        border-radius: 4px;
        color: $dark-gray;
        display: flex;
        margin-bottom: 10px;
        overflow: hidden;

        &-picture {
            height: 75px;
            width: 75px;
        }
    }
}