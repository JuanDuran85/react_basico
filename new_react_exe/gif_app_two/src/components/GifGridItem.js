import React from 'react'

export const GifGridItem = ({id, title, url}) => {
    
    return (
        <div>
            <img  className="animate__animated animate__bounceInLeft" src={url} alt={title}/>
            <p className="card-text">{title}</p>
        </div>
    )
}
