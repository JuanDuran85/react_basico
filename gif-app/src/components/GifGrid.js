import React from 'react';
import { useFetchGif } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({categoria}) => {

    const { data: images, loading } = useFetchGif(categoria);

/*     const [images, setImages] = useState([]);

    useEffect(()=>{
        getGifs(categoria)
            .then(setImages)
            .catch(error => console.error(error));
    },[categoria]); */

    return (
        <>
            <h1 className="animate__animated animate__flash">{categoria}</h1>
            { loading && 'Cargando...'}
            <div className="cardGrid">
                {
                    images.map((img) => (
                        <GifGridItem key={img.id} {...img}/>
                    ))
                }
            </div>
        </>
    );
};

GifGrid.prototype = {
    categoria: PropTypes.string.isRequired,
};