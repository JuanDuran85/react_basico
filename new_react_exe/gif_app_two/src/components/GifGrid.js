import React, { useEffect, useState } from "react";
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ categoria }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifsImg();
  }, []);

  const getGifsImg = async () => {
    const urlApi = `https://api.giphy.com/v1/gifs/search?q=homero&limit=10&api_key=JKJeb3Vwarxcy3pWRoiFYKOgFFjz7epq`;
    const response = await fetch(urlApi);
    const { data } = await response.json();

    const gifsIndvidual = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    setImages(gifsIndvidual);
  };

  return (
    <>
      <h3>{categoria}</h3>
      <ol>
        {images.map(img => (
          <GifGridItem key={img.id} {...img}/>
        ))}
      </ol>
    </>
  );
};
