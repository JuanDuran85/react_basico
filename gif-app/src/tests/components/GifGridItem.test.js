import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruenas en GifGridItem', () => {
    const id = 1;
    const title = "nueva imagen";
    const url = "https://loquesea.com/loquesea.jpg";
    let wrapper = shallow(<GifGridItem id={id} title={title} url={url}/>);
    
    beforeEach(() => {
        wrapper = shallow(<GifGridItem id={id} title={title} url={url}/>);
    });

    test('pruebas basica 1', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe tener un parrafo con el title', () => {
        const encontrado = wrapper.find('p').text().trim();
        expect(encontrado).toBe(title);
    });

    test('imagen igual al url de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('title')).toBe(title);
    });
    
    test('debe tener clase animate__swing', () => {
        const div = wrapper.find('div');
        expect(div.hasClass('animate__swing')).toBe(true);
        const div2 = wrapper.find('div').prop('className');
        expect(div2.includes('animate__swing')).toBe(true);
    });
});