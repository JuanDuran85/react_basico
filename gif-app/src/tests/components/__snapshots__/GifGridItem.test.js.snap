// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruenas en GifGridItem pruebas basica 1 1`] = `
<div
  className="card animate__animated animate__swing"
>
  <img
    alt={1}
    src="https://loquesea.com/loquesea.jpg"
    title="nueva imagen"
  />
  <p>
    nueva imagen
  </p>
</div>
`;
